httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java

```diff
@@ -272,6 +274,20 @@ public SSLContext build() throws NoSuchAlgorithmException, KeyManagementExceptio
         return sslcontext;
     }
 
+    public SSLContext buildWithProvider(final Provider provider) throws NoSuchAlgorithmException, KeyManagementException {
```

Comment:
Do it really take a special `#build` method? Can't we have a `#useProvider` setter similar to `#useProtocol` ?